### DISCUSSION TITLE
getting "origin not allowed" when accessing grafana via teleport proxy

### DISCUSSION THREAD

**Discussion by Blessings9** [gravitational/teleport#32345] [Q&A] [answered]
hello folks
relatively new to Teleport so forgive me if this is not the best forum for this...but as as from the title, i get that error right on the login page and failing to login, i have specified the domain for the application both in the Grana's config file and Teleport's  App Service file...FYI i have installed Grafana 10.1.2 and Teleport v13.3.8 on Ubuntu 22.04.
much appreciated for any assistance

**Comment by scottpgallagher** [comment, marked as answer]
You probably need to inject host and origin headers in your teleport.yaml file where the app_service is running:

rewrite:
  headers:
  - "Host: grafana.teleport.example.com"
  - "Origin: https://grafana.teleport.example.com/"

**Reply by Blessings9** [reply]
Thank you very much @scottpgallagher this resolved the issue

**Reply by pac1f1ste** [reply]
Many thanks buddy! You saved my day :)
